mmengine/model/wrappers/test_time_aug.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from abc import abstractmethod
+from typing import Dict, List, Optional, Union
+
+import torch
+
+from mmengine import MODELS
+from mmengine.optim import OptimWrapper
+from mmengine.structures import BaseDataElement
+from ..base_model import BaseDataPreprocessor, BaseModel
+
+EnhancedInputs = List[Union[torch.Tensor, List[torch.Tensor]]]
+EnhancedDataSamples = List[List[BaseDataElement]]
+MergedDataSamples = List[BaseDataElement]
+
+
+@MODELS.register_module()
+class BaseTestTimeAugModel(BaseModel):
+    """Base model for test time augmentation.
+
+    ``BaseTestTimeAugModel`` is a wrapper for specific algorithm
+    model. It implements the :meth:`forward` for multi-batch
+    data inference. `multi-batch` data means different enhanced results for
```

Comment:
different enhanced results -> data processed by different augmentation from the same batch.